src/core/utils/hooks/useOverflow.tsx

```diff
@@ -60,21 +69,26 @@ export const useOverflow = <T extends HTMLElement>(
     const availableWidth = containerRef.current.offsetWidth;
     const requiredWidth = containerRef.current.scrollWidth;
 
-    // hide items when there's no space available
-    if (availableWidth < requiredWidth && visibleCount > 1) {
-      setVisibleCount((count) => count - 1);
-      overflowBreakpoints.current.push(availableWidth);
-    }
-    // restore items when there's enough space again
-    else if (
-      overflowBreakpoints.current.length > 0 &&
-      availableWidth >
-        overflowBreakpoints.current[overflowBreakpoints.current.length - 1]
-    ) {
-      setVisibleCount((count) => count + 1);
-      overflowBreakpoints.current.pop();
+    if (availableWidth < requiredWidth) {
+      const avgItemWidth = requiredWidth / visibleCount;
+      const visibleItems = Math.floor(availableWidth / avgItemWidth);
+      setVisibleCount(visibleItems);
+    } else if (needsFullRerender.current) {
+      const childrenWidth = Array.from(containerRef.current.children).reduce(
+        (sum: number, child: HTMLElement) => sum + child.offsetWidth,
+        0,
+      );
```

Comment:
If you add type to the array, I think you won't need them in the in callback.
```suggestion
      const childrenWidth = Array.from<HTMLElement>(containerRef.current.children).reduce(
        (sum, { offsetWidth }) => sum + offsetWidth,
        0,
      );
```

Reply:
I tried that previously:
![image](https://user-images.githubusercontent.com/36186912/141740673-baf8b580-f97b-4a3a-81d1-a59bf0468541.png)